compiler/src/lowering/mappers/module.rs

```diff
@@ -15,7 +26,57 @@ pub fn module(context: &Context, module: fe::Module) -> fe::Module {
             fe::ModuleStmt::SimpleImport { .. } => stmt,
             fe::ModuleStmt::ContractDef { .. } => contracts::contract_def(context, stmt),
         })
+        .collect::<Vec<_>>();
+
+    let attributes = context.get_module().expect("missing attributes");
+
+    let tuple_defs = attributes
+        .tuples_used
+        .iter()
+        .map(|tuple| Node::new(tuple_to_struct_def(tuple), Span::new(0, 0)))
+        .collect::<Vec<Node<fe::ModuleStmt>>>();
+
+    fe::Module {
+        body: [tuple_defs, lowered_body].concat(),
+    }
+}
+
+fn tuple_to_struct_def(tuple: &Tuple) -> fe::ModuleStmt {
+    let fields = tuple
+        .items
+        .iter()
+        .enumerate()
+        .map(|(index, typ)| {
+            Node::new(
+                build_struct_field(format!("item{}", index), fixed_size_to_type_desc(typ)),
+                Span::new(0, 0),
+            )
+        })
         .collect();
 
-    fe::Module { body: lowered_body }
+    fe::ModuleStmt::StructDef {
+        name: Node::new("TUPLE_U256_BOOL".to_string(), Span::new(0, 0)),
+        fields,
+    }
+}
+
+fn build_struct_field(name: String, type_desc: fe::TypeDesc) -> fe::Field {
+    fe::Field {
+        pub_qual: None,
+        const_qual: None,
+        name: Node::new(name, Span::new(0, 0)),
+        typ: Node::new(type_desc, Span::new(0, 0)),
+    }
+}
+
+fn fixed_size_to_type_desc(typ: &FixedSize) -> fe::TypeDesc {
+    match typ {
```

Comment:
implement the rest of the base types

Reply:
dont actually. implement some sort of safe_name function on all types and remove the safe_name function from the AbiEncoding trait

Reply:
take care of this in the names file